File: client/public/src/common/model/skills.js
define(function (require, exports, module) {

    const Types = {
        ATTACK: 0,
        BUILD: 1,
        GATHER: 2,
        CRAFT: 3,
        USE: 4,
        SPECIAL: 5
    };

    const Targets = {
        ENEMIES: 0,
        USERS: 1
    };

    const Ids = {
        PUNCH: 0,
        BOW_SHOT: 1,
        SWORD_HIT: 2,
        CREATE_TREE: 3,
        CUT_TREE: 4,
        CREATE_ARROWS: 5,
        INTRODUCE: 6,
        GRAB_APPLE: 7,
        EAT_APPLE: 8,
    };

    module.exports = {
        Types,
        Ids,
        Targets,
        keyToId(skillKey) {
            const skillIdName = skillKey.toUpperCase().replace('-', '_');
            const skillId = Ids[skillIdName];
            if (skillId == null) {
                throw `There is no skill id for a skill ${skillKey}`
            }
            return skillId
        }
    };
});

File: server/server-app/src/main/java/dzida/server/app/instance/skill/Skills.java
package dzida.server.app.instance.skill;

import dzida.server.app.basic.entity.Id;

public class Skills {
    public static class Types {
        public final static int ATTACK = 0;
        public final static int BUILDING = 1;
        public final static int GATHER = 2;
    }

    public static class Target {
        public final static int ENEMIES = 0;
    }

    public static class Ids {
        public final static Id<Skill> PUNCH = new Id<>(0);
        public final static Id<Skill> BOW_SHOT = new Id<>(1);
        public final static Id<Skill> SWORD_HIT = new Id<>(2);
    }
}
